### DISCUSSION TITLE
Problems regarding the Crocus output variables

### DISCUSSION THREAD

**Discussion by Lise-Li** [NCAR/wrf_hydro_nwm_public#784] [Q&A] [answered]
Hi, I'm so excited that the WRF-Hydro model can consider the glacier melt. 
I ran the WRF-Hydro model with Crocus model for 15years, but I found that the meaning of variables related to Crocus are confused since there are no detailed description of these variables. It seemed that all variables related to Crocus considered snow and glacier. I wonder how can I calculate the glacier thickness change only, without considering snow depth? And how can I seperate the glacier melt from total streamflow (is the FLOW_ICE represent the glacier melt?)? Would you mind giving a detailed description of these variables related to Crocus (e.g., PSNOWHEIGHT, PSNOWDZ, FLOW_ICE, FLOW_SNOW)  to isolate the contribution of snow and glaciers to these variables  and the relationship among these variables? 
Looking forward to your reply! Thank you for your kindly help!
Best wishes,
Lise

**Comment by aubreyd** [comment]
Hi Lise-Li: A description of the Crocus implementation and the different variables will be included in our upcoming code release. In the meantime, @trudeeidhammer may be able to answer your questions.

**Comment by trudeeidhammer** [comment, marked as answer]
Hi Lise-Li

I will try and give you some information regarding the outputs:

FLOW_ICE is the streamflow from the Crocus, when the top grid is represented by ice (density > 850 kg/m3) and FLOW_SNOW is the streamflow from the Crocus when the top gridpoint is represented by snow. However, it does not distinguish if this streamflow is from precipitation (rain) or from melting of snow or ice. The way I am attempting to estimate streamflow from snow or ice melt is to subtract the incoming precipitation. 

To calculate glacier thickness change only, I would look at PSNOWRHO. If the density is > 850 kg/m3, the grid cell is considered to be ice. You can then mask PSNOWDZ by where PSNOWRHO > 850 kg/m3 and find the change in glacier thickness. PSNOWHEIGHT is the sum of PSNOWDZ over all levels. 

Please let me know if you have any other questions

**Reply by Lise-Li** [reply]
Hi, @trudeeidhammer and @aubreyd. 

Thank you for your detailed reply. Apologize for the inconvenience, I still have some questions. 

1) The FLOW_ICE and FLOW_SNOW are calculated depend on the top grid situation (ice or snow) as you mentioned above, so the FLOW_ICE includes the glacier melt and rainfall, while the FLOW_SNOW includes  snow melt, glacier melt and rainfall? How can I seperate the melting of glacier from FLOW_SNOW when the top grid is coverd by snow? Can I seperate the melting of snow and ice through calculating the change of ice water equivalent and snow water equivalent?
I noticed that the meaning of PSNOWTHRUFAL is "Runoff from glacier", is this variable representing the glacier melt only?

2) All these variables we discussed above are outputs directly from Crocus, my second question is how can these flow (e.g., FLOW_ICE and FLOW_SNOW) reflect in the runoff (confluence) of watershed oulet (i.e., how can I calculate the snow melt runoff, glacier melt runoff, rainfall runoff at the outlet of watershed)?

Thank you very much for your help!
Lise

**Comment by trudeeidhammer** [comment]
Hi Aubrey

I answered Lise-Li's question, but I am not sure if I will be notified if
she responds.
Do you mind forwarding any potential response from Lise-Li to me?

Thanks,
Trude

On Tue, Nov 5, 2024 at 8:20 AM Aubrey Dugger ***@***.***>
wrote:

> Hi Lise-Li: A description of the Crocus implementation and the different
> variables will be included in our upcoming code release. In the meantime,
> @trudeeidhammer <https://github.com/trudeeidhammer> may be able to answer
> your questions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/wrf_hydro_nwm_public/discussions/784#discussioncomment-11156238>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD24I5PMVYGUXQXTDNOCJSLZ7DO5PAVCNFSM6AAAAABRGTTLM2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJVGYZDGOA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
*My working day may not be your working day. Please do not feel obliged to
reply to this email outside of your normal working hours.*

She/Her/Hers
Dr. Trude Eidhammer
Project Scientist II
Research Applications Laboratory (RAL)
NSF NCAR
P.O. Box 3000
Boulder, CO 80307-3000, USA
